cpp/bulk_mailer.cc

```diff
@@ -0,0 +1,107 @@
+/** \file    bulk_mailer.cc
+ *  \brief   Sends emails to a list of users.
+ *  \author  Dr. Johannes Ruscheinski
+ */
+
+/*
+    Copyright (C) 2020 Library of the University of Tübingen
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <vector>
+#include <cstdlib>
+#include "DbConnection.h"
+#include "DbResultSet.h"
+#include "DbRow.h"
+#include "EmailSender.h"
+#include "FileUtil.h"
+#include "VuFind.h"
+#include "util.h"
+
+
+namespace {
+
+
+[[noreturn]] void Usage() {
+    ::Usage("mail_contents sql_statement auxillary_email_address_list reply_to_email_address\n"
+            "\tmail_contents: the file containing the email message\n"
+            "\t\t(the first line will be used as the mail's subject)\n"
+            "\tsql_statement: a query to select the relevant email addresses from the vufind user table\n"
```

Comment:
Since CollectRecipientsFromSqlTable returns row[0], it might be a good idea to explicitly say that the given SQL statement may only select 1 column